addons/addon-base-raas/packages/base-raas-services/lib/costs/costs-service.js

```diff
@@ -157,23 +159,34 @@ class CostsService extends Service {
         Filter: filter,
         GroupBy: groupBy,
       })
-      .promise();
-
-    const response = result.ResultsByTime.map(item => {
-      const costItems = {};
-      item.Groups.forEach(group => {
-        if (group.Metrics.BlendedCost.Amount > 0) {
-          costItems[group.Keys] = {
-            amount: Math.round(group.Metrics.BlendedCost.Amount * 100) / 100,
-            unit: group.Metrics.BlendedCost.Unit,
-          };
+      .promise()
+      .catch(e => {
+        // The DataUnavailableException represents non-availability of data for specific cost usage search criteria.
+        // Just return undefined instead of throwing error in this case for easier API usage in the client code.
+        if (e.code === 'DataUnavailableException') {
+          return undefined;
         }
+        throw e;
       });
```

Comment:
Nitpick: since we are using async/await it is weird we are using chaining here. Could we switch to follow try/catch?

Reply:
The following seems more concise and easier to read (arguably :)) where a promise resolves to some result and you want to use that result in further statements. The try/catch is better when you want to skip group of statements. 
```
const someVar = await somePromise.catch(e => // handle e in very specific case but re-throw in other cases );

// bunch of other code using someVar
```

vs 

```
let someVar;
try {
 someVar = await somePromise;
} catch (e) {
 // handle e in very specific case but re-throw in other cases 
}

// bunch of other code using someVar
```

As a by-product you can declare the var `const` instead of declare and reassign. I prefer try/catch in following situation
```
try {
const someVar = await somePromise;

// bunch of other code using someVar
} catch (e) {
// handle e
}
```

But, again the code currently does not have any definitive style guide for such cases and it may be inconsistent on this front.

Reply:
Good points -- we haven't discussed this aspect yet for now we can leave as is